multiapps-controller-persistence/src/main/java/org/cloudfoundry/multiapps/controller/persistence/util/DataSourceFactory.java

```diff
@@ -28,11 +28,29 @@ private HikariConfig createHikariConfig(CfJdbcService service, Integer maximumPo
         hikariConfig.setIdleTimeout(60000);
         hikariConfig.setMinimumIdle(10);
         hikariConfig.addDataSourceProperty("tcpKeepAlive", true);
+
+        configureSSLClientKeyIfExists(service, hikariConfig);
+
         if (maximumPoolSize != null) {
             hikariConfig.setMaximumPoolSize(maximumPoolSize);
         }
         hikariConfig.setRegisterMbeans(true);
         return hikariConfig;
     }
 
+    private void configureSSLClientKeyIfExists(CfJdbcService service, HikariConfig hikariConfig) {
+        Object clientKey = service.getCredentials()
+                                  .getMap()
+                                  .get("sslkey");
+        if (clientKey != null) {
+            configureClientCertificate(clientKey.toString(), hikariConfig);
+        }
+    }
+
+    private void configureClientCertificate(String clientKey, HikariConfig hikariConfig) {
+        ClientKeyConfigurationHandler sslKeyHandler = new ClientKeyConfigurationHandler();
+        sslKeyHandler.createEncodedKeyFile(clientKey, "ClientKey.der");
```

Comment:
Extract the filename as constant.
If this method returns a File, you can use it, this way you will not hardcode paths in the filesystem. Not really recomended